src/main/java/org/apache/accumulo/access/BytesWrapper.java

```diff
@@ -21,12 +21,13 @@
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 import java.util.Arrays;
+import java.util.Objects;
 
 final class BytesWrapper implements Comparable<BytesWrapper> {
 
-  protected byte[] data;
-  protected int offset;
-  protected int length;
+  private byte[] data;
+  private int offset;
+  private int length;
```

Comment:
That raises the question: is the class only intended for use internally where we are using it carefully, or is it intended as part of this library's public API? I ask because it appears to be public API.

Reply:
It is not part of the public API, its package private.  The project has [this script](https://github.com/apache/accumulo-access/blob/main/src/build/ci/find-unapproved-public.sh) that attempts to validate that only a small subset of types are public.  The project only has single package so the non public api stuff is set to package private and then the script checks that. 

I suppose we could also use the new Java module mechanisms for this project since its new and so small, should hopefully be easy to do.

Reply:
Oh, my mistake. For some reason I thought I saw "public class". You can disregard my noise. :smiley_cat:

Reply:
I'd be in favor of making a module for this library.